File: 13. Roman to Integer/solution.cpp
class Solution {
public:
    int romanToInt(string s) {
        int res = 0;
        
        map<char, int> d;
        d = {{'I', 1}, {'V', 5}, {'X',10}, {'L',50}, {'C',100}, {'D',500}, {'M',1000}};
        
        int i = 0;
        while(i < s.size()){
            if((i != s.size() - 1) && (d[s[i]] < d[s[i+1]])){
                res += d[s[i+1]] - d[s[i]];
                i += 2;
            }
            else{res += d[s[i]];
                i ++;}
        }
        return res;
    }
};

File: 13. Roman to Integer/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mapp = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        if len(s) == 1: return mapp[s[0]]
        res = 0
        idx = 0
        while idx < len(s):
            if idx != len(s) - 1 and mapp[s[idx]] < mapp[s[idx+1]]:
                res += mapp[s[idx+1]] - mapp[s[idx]]
                idx += 2
            else:
                res += mapp[s[idx]]
                idx += 1
        return res
